lib/connection.ml

```diff
@@ -140,8 +140,9 @@ let connect ~sw ~clock ~network ~config conn_info =
   with
   | Ok sock -> Ok sock
   | Error `Timeout
+  | exception Eio.Exn.Io _
```

Comment:
instead of catching all `Eio.Exn.Io` errors, shouldn't we just be handling `Eio.Exn.Io (E Eio.Net.Connection_failuire`?

see example in https://github.com/anmonteiro/gluten/pull/50

Reply:
Updated to let `Connection_reset` fallthrough.